### PR TITLE
Update doctest to v2.4.9

### PR DESCRIPTION
**Pull request by Saalvage**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **Yes** (carefully enough to find a typo in it, the fix of which I have taken the liberty of including in this pull request. If this is an issue, please tell me and I'll force-push it out, personally not a fan of single character PRs!)

* This update will fix a few toolchains.
No toolchains appear to be broken.
Yet the currently run toolchains seem to differ, both from the ones used in #353 as well as the ones in https://cpp-pm.github.io/hunter/, so I am unable to name the fixed toolchains exactly.

---

### CODE REVIEW

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -257,7 +257,7 @@ hunter_default_version(dfdutils VERSION 0.0.0-659a739-p1)
 hunter_default_version(dlib VERSION 19.17-p0)
 hunter_default_version(dlpack VERSION 0.2-0acb731)
 hunter_default_version(dmlc-core VERSION 0.3-3943914-p0)
-hunter_default_version(doctest VERSION 2.4.4)
+hunter_default_version(doctest VERSION v2.4.9)
```
Comment (NeroBurner) [inline cmake/configs/default.cmake:260]: the other packages don't use the leading `v` in the version number. The `v` isn't technically part of the version number generally

```suggestion
hunter_default_version(doctest VERSION 2.4.9)
```

File: cmake/projects/doctest/hunter.cmake [hunk 2]
```diff
@@ -65,6 +65,17 @@ hunter_add_version(
     d1baae099a165b231bb4fdd3637789f7465d873a
 )
 
+hunter_add_version(
+    PACKAGE_NAME
+    doctest
+    VERSION
+    "v2.4.9"
```
Comment (NeroBurner) [inline cmake/projects/doctest/hunter.cmake:72]:
```suggestion
    "2.4.9"
```

File: docs/creating-new/test.rst [hunk 3]
```diff
@@ -8,7 +8,7 @@ Hunter uses `GitHub Actions <https://docs.github.com/en/actions>`__ for :ref:`co
 You can also test :ref:`package building <testing locally>` and :ref:`documentation <testing documentation locally>` locally, however this is optional.
 
 Testing will be performed automatically on pull request.
-To perform testing on push to yours Hunter fork, ensure that GitHub Actions are enabled for your repository -
+To perform testing on push to your Hunter fork, ensure that GitHub Actions are enabled for your repository -
```
Comment (NeroBurner) [inline docs/creating-new/test.rst:11]: please open a new PR for this change

```suggestion
To perform testing on push to yours Hunter fork, ensure that GitHub Actions are enabled for your repository -
```

**Review by NeroBurner — CHANGES_REQUESTED** [review]

File: cmake/configs/default.cmake (hunk 1 above)
Comment (Saalvage) [reply cmake/configs/default.cmake]: Some other packages seem to use it so I assumed it was dependent on the versioning scheme used by the project proper, thanks!

**Review by NeroBurner — APPROVED** [review]
LGTM now

I was a bit puzzled by the URL change from `onqtam` to `doctest`, but the `onqtam` URL now forwards to `doctest`, so all seems well

**Pull request merged**